src/pupil_labs/invisible_lsl_relay/outlets.py

```diff
@@ -38,37 +48,45 @@ def push_sample_to_outlet(self, sample):
 
 
 class PupilInvisibleGazeOutlet(PupilInvisibleOutlet):
-    def __init__(self, device_id=None):
+    def __init__(self, device_id, channel_prefix=None):
         PupilInvisibleOutlet.__init__(
             self,
             channel_func=pi_gaze_channels,
-            outlet_name='Gaze',
+            outlet_type='Gaze',
             outlet_format=lsl.cf_double64,
             timestamp_query=pi_extract_from_sample('timestamp_unix_seconds'),
-            outlet_uuid=f'{device_id or str(uuid.uuid4())}_Gaze',
+            channel_name_prefix=channel_prefix,
+            outlet_uuid=f'{device_id}_Gaze',
         )
 
 
 class PupilInvisibleEventOutlet(PupilInvisibleOutlet):
-    def __init__(self, device_id=None):
+    def __init__(self, device_id, channel_prefix=None):
         PupilInvisibleOutlet.__init__(
             self,
             channel_func=pi_event_channels,
-            outlet_name='Event',
+            outlet_type='Event',
             outlet_format=lsl.cf_string,
             timestamp_query=pi_extract_from_sample('timestamp_unix_seconds'),
-            outlet_uuid=f'{device_id or str(uuid.uuid4())}_Event',
+            channel_name_prefix=channel_prefix,
+            outlet_uuid=f'{device_id}_Event',
         )
 
 
-def pi_create_outlet(outlet_uuid, channels, outlet_name, outlet_format):
-    stream_info = pi_streaminfo(outlet_uuid, channels, outlet_name, outlet_format)
+def pi_create_outlet(
+    outlet_uuid, channels, outlet_type, outlet_format, channel_name_prefix
+):
+    stream_info = pi_streaminfo(
+        outlet_uuid, channels, outlet_type, outlet_format, channel_name_prefix
+    )
     return lsl.StreamOutlet(stream_info)
 
 
-def pi_streaminfo(outlet_uuid, channels, type_name: str, channel_format):
+def pi_streaminfo(
+    outlet_uuid, channels, type_name: str, channel_format, channel_name_prefix
+):
     stream_info = lsl.StreamInfo(
-        name=f"pupil_invisible_{type_name}",
+        name=f"{channel_name_prefix}_{type_name}",
```

Comment:
I fear that the variable name is misleading. This sets the _outlet_ name, not the _channel_ name. Each outlet can have multiple channels, each with its own name. But these are not modified by this prefix. Please adjust the variable name accordingly throughout this PR.